pdfiumandroid/src/main/cpp/pdfiumandroid.cpp

```diff
@@ -79,6 +108,14 @@ DocumentFile::~DocumentFile(){
         free(cDataCopy);
         cDataCopy = nullptr;
     }
+    if(nativeSourceBridgeGlobalRef != nullptr){
+        JNIEnv *env;
+        bool attached;
+        if(jniAttachCurrentThread(&env, &attached)){
+            env->DeleteGlobalRef(nativeSourceBridgeGlobalRef);
```

Comment:
And here's the reference cleanup in destructor what also uses JNI attach/detach to obtain valid JNIEnv.